src/script/view_model/ContentViewModel.ts

```
@@ -218,8 +219,25 @@ export class ContentViewModel {
     );
   }
 
-  private isConversationNotFoundError(error: any): boolean {
-    return error.type === ConversationError.TYPE.CONVERSATION_NOT_FOUND;
+  private isConversationNotFoundError(error: unknown): boolean {
+    return isError(error) && 'type' in error && error.type === ConversationError.TYPE.CONVERSATION_NOT_FOUND;
```

same question, we are assuming error is an error instance, will it be always true?

---

In this function? No, it depends on what you give it. But this typeguard has one job only, check if whatever you give it is a ConversationNotFoundError.

---

my question was is it possible a non error instance arrives here then isError(error) will fail and rest of the condition won't be checked right?

---

This is a typeguard. It is a pure function that has one job only and doesnt care about anything else. 

How the case is handled when it returns false, it doesn't know and it is the job of the caller to implement. 

In this case, we only want to show a modal if the error type matches; otherwise, we ignore the error. My refactor just fixed the typeguard implementation, not the logic around it